tests/test-distribution/test-distribution-common/src/test/java/org/eclipse/jetty/tests/distribution/StatsTests.java

```diff
@@ -83,87 +69,10 @@ public void testStatsServlet(String env) throws Exception
-                assertEquals(HttpStatus.OK_200, response.getStatus());
-
-                assertThat("Response.contentType", response.getHeaders().get(HttpHeader.CONTENT_TYPE), containsString("application/json"));
-
-                Object doc = new JSON().parse(new JSON.StringSource(response.getContentAsString()));
-                assertNotNull(doc);
-                assertThat(doc, instanceOf(Map.class));
-                Map<?, ?> docMap = (Map<?, ?>)doc;
-                assertEquals(4, docMap.size());
-                assertNotNull(docMap.get("requests"));
-                assertNotNull(docMap.get("responses"));
-                assertNotNull(docMap.get("connections"));
-                assertNotNull(docMap.get("memory"));
-
-                // ---------------
-                // Test TEXT accept
-                response = client.newRequest(serverBaseURI.resolve("/demo/stats"))
-                    .method(HttpMethod.GET)
-                    .headers((headers) -> headers.add(HttpHeader.ACCEPT, "text/plain"))
-                    .send();
-                assertEquals(HttpStatus.OK_200, response.getStatus());
-
-                assertThat("Response.contentType", response.getHeaders().get(HttpHeader.CONTENT_TYPE), containsString("text/plain"));
-
-                String textContent = response.getContentAsString();
-                assertThat(textContent, containsString("requests: "));
-                assertThat(textContent, containsString("responses: "));
-                assertThat(textContent, containsString("connections: "));
-                assertThat(textContent, containsString("memory: "));
-
-                // ---------------
-                // Test HTML accept
-                response = client.newRequest(serverBaseURI.resolve("/demo/stats"))
-                    .method(HttpMethod.GET)
-                    .headers((headers) -> headers.add(HttpHeader.ACCEPT, "text/html"))
-                    .send();
-                assertEquals(HttpStatus.OK_200, response.getStatus());
-
-                assertThat("Response.contentType", response.getHeaders().get(HttpHeader.CONTENT_TYPE), containsString("text/html"));
-
-                String htmlContent = response.getContentAsString();
-                // Look for things that indicate it's a well formed HTML output
-                assertThat(htmlContent, containsString("<html>"));
-                assertThat(htmlContent, containsString("<body>"));
-                assertThat(htmlContent, containsString("<em>requests</em>: "));
-                assertThat(htmlContent, containsString("<em>responses</em>: "));
-                assertThat(htmlContent, containsString("<em>connections</em>: "));
-                assertThat(htmlContent, containsString("<em>memory</em>: "));
-                assertThat(htmlContent, containsString("</body>"));
-                assertThat(htmlContent, containsString("</html>"));
+                // TODO test the StatisticsHandler somehow
```

Comment:
We could make a couple requests, there already is the one above that returns a 200, and add another one that returns a 404, set `dumpBeforeStop` to true, then stop the server and check that the `2xxResponses` and `4xxResponses` counters are part of the dump and have the expected values?

Reply:
yeah, but would be good to see requests in progress.